x/programs/rust/wasmlanche-sdk/src/memory.rs

```diff
@@ -16,24 +16,6 @@ thread_local! {
     static GLOBAL_STORE: RefCell<HashMap<*const u8, usize>> = RefCell::new(HashMap::new());
 }
 
-/// Converts a pointer to a i64 with the first 4 bytes of the pointer
-/// representing the length of the memory block.
-/// # Errors
-/// Returns an [`StateError`] if the pointer or length of `args` exceeds
-/// the maximum size of a u32.
```

Comment:
If this function is ever coming back again, we should make sure to write a correct comment

Reply:
I don't think it's ever coming back. The current version of `wasmtime` we're using only supports `size_of::<usize>() == size_of::<u32>()`, so the safety condition here isn't relevant. 

If there comes a point in time when we upgrade to support a 64-bit runtime, we'll have to do something to handle existing user code anyway.